src/core/common/num_utils.hpp

```diff
@@ -131,6 +131,27 @@ template <typename UintType> uint16_t ClampToUint16(UintType aValue)
     return static_cast<uint16_t>(Min(aValue, static_cast<UintType>(NumericLimits<uint16_t>::kMax)));
 }
 
+/**
+ * This template function returns a clamped version of given integer to a `int8_t`.
+ *
+ * If @p aValue is smaller than min value of a `int8_t`, the min value of `int8_t` is returned.
+ *
+ * @tparam IntType    The value type (MUST be `int16_t`, `int32_t`, or `int64_t`).
+ *
+ * @param[in] aValue  The value to clamp.
+ *
+ * @returns The clamped version of @p aValue to `int8_t`.
+ *
+ */
+template <typename IntType> int8_t ClampToInt8L(IntType aValue)
+{
+    static_assert(TypeTraits::IsSame<IntType, int16_t>::kValue || TypeTraits::IsSame<IntType, int32_t>::kValue ||
+                      TypeTraits::IsSame<IntType, int64_t>::kValue,
+                  "IntType must be `int16_t, `int32_t`, or `int64_t`");
+
+    return static_cast<int8_t>(Max(aValue, static_cast<IntType>(NumericLimits<int8_t>::kMin)));
```

Comment:
Thanks for adding this.
I think better to add check for max value as well (since `Clamp` generally means to limits to a min-max range) - not just `L` side.
We can use existing `Clamp()` function here?
```suggestion
    return static_cast<int8_t>(Clamp(aVlue, static_cast<IntType>(NumbericLimits<int8_t>::kMin, static_cast<IntType>(NumbericLimits<int8_t>::kMax);
```

Reply:
SGTM! Updated.